netket/graph/lattice.py

```diff
@@ -19,9 +19,17 @@
 import itertools
 import networkx as _nx
 import warnings
+from typing import Tuple, Union, Optional
+
+cutoff_tol = 1e-5
+"""Tolerance for the maximum distance cutoff when computing the sparse distance matrix.
+This is necessary because of floating-point errors when computing the distance in non-trivial 
+lattices.
+"""
 
 
 def get_edges(atoms_positions, cutoff):
+    cutoff = cutoff + cutoff_tol
     kdtree = cKDTree(atoms_positions)
     dist_matrix = kdtree.sparse_distance_matrix(kdtree, cutoff)
```

Comment:
it's really unfortunate that ```sparse_distance_matrix``` does not have a ```tol``` parameter...  I am pretty sure that they have similar issue internally if the cutoff is set up to be too close to the tolerance!

Reply:
I opened an [issue](https://github.com/scipy/scipy/issues/13924). Maybe in the future they'll take a look at this.